src/Symfony/Component/TypeInfo/IntRangeBoundBuiltinType.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo;
+
+/**
+ * Int range bound type.
+ *
+ * @author Martin Rademacher <mano@radebatz.net>
+ *
+ * @extends BoundBuiltinType<BoundBuiltinTypeIdentifier::RANGE_INT>
+ */
+class IntRangeBoundBuiltinType extends BoundBuiltinType
```

Comment:
```suggestion
final class IntRangeType extends BuiltinType
```

I'd rather make the `BuiltinType` non-final, and then created a `IntRangeType` instead. It'll make it more logic, as an integer range is indeed a subtype of integer.

If in the future there are other subtypes, such as `key-of`, `value-of`, `class-string`, ... (see https://phpstan.org/writing-php-code/phpdoc-types), I think it'll be clearer to create a dedicated class for each use case.

Reply:
Yeah, I thought about that. If we do go down that path, how would you see simple bound type enum values (for example `html-escaped-string`) be attached to builtin types in general? Optionally passed in directly into the constructor? (removing the need for my `BoundBuiltinType`)- something like:
```php
// TypeFactoryTrait

    /**
     * @return BuiltinType<TypeIdentifier::STRING>
     */
    public static function string(?BoundBuiltinTypeIdentifier $boundBuiltinType = null): BuiltinType
    {
        return self::builtin(TypeIdentifier::STRING, $boundBuiltinType);
    }
```